### DISCUSSION TITLE
Velero Uninstall: Velero namespace stuck in "Terminating" state

### DISCUSSION THREAD

**Discussion by gh-noobie** [vmware-tanzu/velero#8379] [Community support Q&A]
Hello everyone,

After uninstalling Velero, all resources were deleted, but the Velero namespace is stuck in a "Terminating" state. Has anyone encountered this before? If so, could you please share any solutions or workarounds that worked for you? I tried manually deleting the namespace with some commands, but it didn't work.
Velero version: v1.14.1
I tried these commands, but they didn't work:
`velero uninstall`
`kubectl delete namespace velero --grace-period=0 --force`
`kubectl get ns velero -o json | jq '.spec.finalizers = []' | kubectl replace --raw "api/v1/namespaces/velero/finalize" -f -`

Thank you in advance for your help!

**Comment by kaovilai** [comment]
kubectl get ns velero -oyaml

**Reply by kaovilai** [reply]
then we'll be able to see what's blocking.

**Comment by gh-noobie** [comment]
@kaovilai, thank you for your response. I executed `kubectl get ns velero -oyaml`, and here is the result:
> apiVersion: v1
> kind: Namespace
> metadata:
>   creationTimestamp: "2024-11-20T04:37:33Z"
>   deletionTimestamp: "2024-11-20T04:45:54Z"
>   labels:
>     component: velero
>     kubernetes.io/metadata.name: velero
>     pod-security.kubernetes.io/audit: privileged
>     pod-security.kubernetes.io/audit-version: latest
>     pod-security.kubernetes.io/enforce: privileged
>     pod-security.kubernetes.io/enforce-version: latest
>     pod-security.kubernetes.io/warn: privileged
>     pod-security.kubernetes.io/warn-version: latest
>   name: velero
>   resourceVersion: "#######"
>   uid: #######a-a#a#-##a#-a###-##a#a#a#a#
> spec:
>   finalizers:
>   - kubernetes
> status:
>   conditions:
>   - lastTransitionTime: "2024-11-20T04:45:59Z"
>     message: 'Discovery failed for some groups, 1 failing: unable to retrieve the
>       complete list of server APIs: metrics.k8s.io/v1beta1: stale GroupVersion discovery:
>       metrics.k8s.io/v1beta1'
>     reason: DiscoveryFailed
>     status: "True"
>     type: NamespaceDeletionDiscoveryFailure
>   - lastTransitionTime: "2024-11-20T04:45:59Z"
>     message: All legacy kube types successfully parsed
>     reason: ParsedGroupVersions
>     status: "False"
>     type: NamespaceDeletionGroupVersionParsingFailure
>   - lastTransitionTime: "2024-11-20T04:45:59Z"
>     message: All content successfully deleted, may be waiting on finalization
>     reason: ContentDeleted
>     status: "False"
>     type: NamespaceDeletionContentFailure
>   - lastTransitionTime: "2024-11-20T04:45:59Z"
>     message: All content successfully removed
>     reason: ContentRemoved
>     status: "False"
>     type: NamespaceContentRemaining
>   - lastTransitionTime: "2024-11-20T04:45:59Z"
>     message: All content-preserving finalizers finished
>     reason: ContentHasNoFinalizers
>     status: "False"
>     type: NamespaceFinalizersRemaining
>   phase: Terminating

I also tried to install Velero by following the procedure outlined in this documentation:
[Disaster Recovery — Notes on Velero and OKE, Part 2: Stateful Pods with Persistent Volumes and Block Volume](https://medium.com/oracledevs/disaster-recovery-notes-on-velero-and-oke-part-2-stateful-pods-with-persistent-volumes-and-80204b3ac6d7)

**Reply by kaovilai** [reply]
This is not a velero specific issue. I recommend finding ways to get metric server back up.

See: https://stackoverflow.com/a/68319002

Similar to [your error here](https://github.com/vmware-tanzu/velero/discussions/8379#discussioncomment-11320374:~:text=Discovery%20failed%20for%20some%20groups%2C%201%20failing%3A%20unable%20to%20retrieve%20the%0Acomplete%20list%20of%20server%20APIs%3A%20metrics.k8s.io/v1beta1%3A%20stale%20GroupVersion%20discovery%3A%0Ametrics.k8s.io/v1beta1).

**Comment by gh-noobie** [comment]
@kaovilai, thank you for your response. I really appreciate your help! The issue has been resolved. It turns out the problem was that there were no worker nodes, which caused it to get stuck in the "Terminating" state.
